src/services/utils.hooks.js

```diff
@@ -2,54 +2,66 @@ const { NotFound } = require('@feathersjs/errors');
 
 /**
  * Checks if requested user is an owner of the resource.
+ * @param { string } service - name of service resource comes from
+ * @param { string } queryParam - field name which allows to identify resource e.g. _id
+ * @param { string } userParam - field name of user identifier in resource e.g. owner
  */
-async function checkOwner(context, service, query) {
-    const { user } = context.params;
+function checkOwner(service, queryParam, userParam) {
+    return async function (context) {
+        const { user } = context.params;
+        const resourceId = context.arguments[0];
 
-    const resourceId = context.arguments[0];
-    const resources = await context.app.service(service).find({
-        query: query,
-    });
+        const query = {}
+        query[queryParam] = resourceId
 
-    let teacher = undefined;
+        const resources = await context.app.service(service).find({
+            query: query,
+        });
 
-    if (resources.length === 1) {
-         teacher = resources[0].teacher;
-    }
+        let requestedUser = undefined;
```

Comment:
Why not just?:

```javascript
if (resources.length && resources[0][userParam] === user.username) {
    return context;
} else {
    throw new NotFound();
}
```

`.length` can return only an integer and zero is falsy, so no need for `===`. If an object has no
length, `undefined` is falsy too.